graylog2-server/src/main/java/org/graylog2/security/ldap/LdapConnector.java

```diff
@@ -303,6 +305,32 @@ public LdapEntry search(LdapNetworkConnection connection,
         return groups;
     }
 
+    /**
+     * Makes sure the given DN string is normalized so it can be used in string comparison.
+     *
+     * Example:
+     *
+     *   input  = "cn=John Doe, ou=groups, ou=system"
+     *   output = "cn=John Doe,ou=groups,ou=system"
+     *
+     * See: https://github.com/Graylog2/graylog2-server/issues/1790
+     *
+     * @param dn denormalized DN string
+     * @return normalized DN string
+     */
+    private String normalizedDn(String dn) {
+        if (isNullOrEmpty(dn)) {
+            return dn;
+        } else {
+            try {
+                return new Dn(dn).getNormName();
+            } catch (LdapInvalidDnException e) {
+                LOG.warn("Invalid DN", e);
+                return dn;
```

Comment:
I didn't want to change the current behavior in case of an invalid DN. If we throw an exception here, we have to handle it somewhere or let it bubble up.